From 302bb9510ecc6093c579447f1986de337907a15e Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Tue, 12 Jun 2018 12:25:47 -0700
Subject: [PATCH 1683/2367] ASoC: SOF: topology: fix dmic pdm token offsets

This patch fixes the dmic pdm token offsets to reflect the correct
offset within the sof_ipc_dai_dmic_pdm_ctrl structure.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 sound/soc/sof/topology.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/sound/soc/sof/topology.c b/sound/soc/sof/topology.c
index dfaae19..89428f1 100644
--- a/sound/soc/sof/topology.c
+++ b/sound/soc/sof/topology.c
@@ -460,31 +460,31 @@ static const struct sof_topology_token dmic_tokens[] = {
 static const struct sof_topology_token dmic_pdm_tokens[] = {
 	{SOF_TKN_INTEL_DMIC_PDM_CTRL_ID,
 		SND_SOC_TPLG_TUPLE_TYPE_SHORT, get_token_u16,
-		offsetof(struct sof_ipc_dai_dmic_params, pdm[0]),
+		offsetof(struct sof_ipc_dai_dmic_pdm_ctrl, id),
 		0},
 	{SOF_TKN_INTEL_DMIC_PDM_MIC_A_Enable,
 		SND_SOC_TPLG_TUPLE_TYPE_SHORT, get_token_u16,
-		offsetof(struct sof_ipc_dai_dmic_params, pdm[0]),
+		offsetof(struct sof_ipc_dai_dmic_pdm_ctrl, enable_mic_a),
 		0},
 	{SOF_TKN_INTEL_DMIC_PDM_MIC_B_Enable,
 		SND_SOC_TPLG_TUPLE_TYPE_SHORT, get_token_u16,
-		offsetof(struct sof_ipc_dai_dmic_params, pdm[0]),
+		offsetof(struct sof_ipc_dai_dmic_pdm_ctrl, enable_mic_b),
 		0},
 	{SOF_TKN_INTEL_DMIC_PDM_POLARITY_A,
 		SND_SOC_TPLG_TUPLE_TYPE_SHORT, get_token_u16,
-		offsetof(struct sof_ipc_dai_dmic_params, pdm[0]),
+		offsetof(struct sof_ipc_dai_dmic_pdm_ctrl, polarity_mic_a),
 		0},
 	{SOF_TKN_INTEL_DMIC_PDM_POLARITY_B,
 		SND_SOC_TPLG_TUPLE_TYPE_SHORT, get_token_u16,
-		offsetof(struct sof_ipc_dai_dmic_params, pdm[0]),
+		offsetof(struct sof_ipc_dai_dmic_pdm_ctrl, polarity_mic_b),
 		0},
 	{SOF_TKN_INTEL_DMIC_PDM_CLK_EDGE,
 		SND_SOC_TPLG_TUPLE_TYPE_SHORT, get_token_u16,
-		offsetof(struct sof_ipc_dai_dmic_params, pdm[0]),
+		offsetof(struct sof_ipc_dai_dmic_pdm_ctrl, clk_edge),
 		0},
 	{SOF_TKN_INTEL_DMIC_PDM_SKEW,
 		SND_SOC_TPLG_TUPLE_TYPE_SHORT, get_token_u16,
-		offsetof(struct sof_ipc_dai_dmic_params, pdm[0]),
+		offsetof(struct sof_ipc_dai_dmic_pdm_ctrl, skew),
 		0},
 };
 
-- 
2.7.4

